src/test/java/bio/terra/service/job/StairwayExceptionSerializerTest.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.service.job;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.app.configuration.ApplicationConfiguration;
+import bio.terra.buffer.client.ApiException;
+import bio.terra.common.category.Unit;
+import bio.terra.service.resourcemanagement.exception.BufferServiceAPIException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles({"google", "unittest"})
+@Tag(Unit.TAG)
+class StairwayExceptionSerializerTest {
+
+  private StairwayExceptionSerializer stairwayExceptionSerializer;
+
+  private String serializedException;
+  private Exception originalException;
+
+  @BeforeEach
+  void setUp() {
+    ObjectMapper objectMapper = new ApplicationConfiguration().objectMapper();
+    stairwayExceptionSerializer = new StairwayExceptionSerializer(objectMapper);
+
+    originalException =
+        new BufferServiceAPIException(
+            new ApiException(
+                "No projects are available in the datarepo pool",
+                HttpStatus.NOT_FOUND.value(),
+                null,
+                null));
+    // Fake providing step info in stack trace
+    var fakeStackTrace =
+        new StackTraceElement(
+            "bio.terra.common.GetResourceBufferProjectStep",
+            "handoutResource",
+            "BufferService.java",
+            83);
+    originalException.setStackTrace(new StackTraceElement[] {fakeStackTrace});
```

Comment:
Instead of making this test depend on `GetResourceBufferProjectStep`, which may change or be moved in the future, you can instead use `Step` directly to satisfy the validation in `StairwayExceptionFieldsFactory`. It's minor but I would also create the trace as an array at declaration to avoid having to call `new`
```suggestion
    StackTraceElement[] fakeStackTrace = {
      new StackTraceElement(Step.class.getName(), "method", "file", 0)
    };
    originalException.setStackTrace(fakeStackTrace);
```